projects/sbb-esta/angular-business/src/lib/header/header/header.component.ts

```diff
@@ -64,51 +59,38 @@ export class HeaderComponent implements OnInit {
   @ViewChild('navigationButtons', { static: true, read: ViewContainerRef })
   navigationButtons: ViewContainerRef;
 
+  /**
+   * Reference to icon, if given through projection.
+   */
   @ViewChild('iconContent', { static: true })
   iconContent: ElementRef;
 
-  private _buttonSpacing = 70;
-  private _componentFactory: ComponentFactory<NavbuttonComponent>;
-  private _left = 0;
+  /**
+   * Distance between navigation buttons, handled through code.
+   */
+  buttonSpacing = 70;
 
-  constructor(private _resolver: ComponentFactoryResolver) {}
+  /** @docs-private */
+  private _left = 0;
 
   ngOnInit() {
     this._checkLabel();
-    const element = this.ngContent.nativeElement;
-    this._componentFactory = this._resolver.resolveComponentFactory(NavbuttonComponent);
+  }
 
-    while (element.children.length > 0) {
-      this._addElementToNavbar(element.children[0]);
+  ngAfterViewInit() {
+    // Absolute positioning of buttons so that they're all 70px apart and won't
```

Comment:
I'm still thinking about an alternative for this, but I think that would require a change to the dropdown implementation...
It's fine for now